src/shims/intrinsics.rs

```diff
@@ -19,12 +16,11 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         span: Span,
         instance: ty::Instance<'tcx>,
         args: &[OpTy<'tcx, Tag>],
-        dest: PlaceTy<'tcx, Tag>,
+        dest: Option<PlaceTy<'tcx, Tag>>,
+        _ret: Option<mir::BasicBlock>,
+        unwind: Option<mir::BasicBlock>
     ) -> InterpResult<'tcx> {
         let this = self.eval_context_mut();
-        if this.emulate_intrinsic(span, instance, args, dest)? {
-            return Ok(());
-        }
```

Comment:
Why did you move this down?

Reply:
At one point, `emulate_intrinsic` was intercepting something that we needed to handle. This doesn't seem to be the case anymore.

Reply:
Actually, this is still needed - `emulate_intrinsic` handles non-diverging intrinsics, so we need to call it after we handle the diverging ones.

Reply:
Hm, I think we should change that then. The rustc_mir stuff should always take precedence over what we have here, so it should be called first.

`emulate_intrinsic` should just return `false` on diverging intrinsics if it doesn't emulate any of them.

Reply:
Sounds good.